core/src/main/java/hudson/model/Queue.java

```diff
@@ -1560,6 +1560,34 @@ public void maintain() {
         }
     }
 
+    private Runnable buildOnTemporaryNode(final BuildableItem p){
```

Comment:
what makes this a "temporary Node"?  (method naming police calling)

The code looks like it can only handle flyweight tasks - but there is no documentation or checks to state/validate this.